src/embed.js

```diff
@@ -68,11 +76,25 @@ function embed(el, opt, callback) {
       source = JSON.stringify(spec, null, 2);
       opt = {spec: spec, actions: false};
     }
-    mode = MODES[opt.mode] || MODES.vega;
+
+    // Decide mode
+    if (opt.mode) {
```

Comment:
We should throw a warning if the mode from opt is not the same as the one in the schema url.